File: Stats.Delphi/List_Func.pas
unit List_Func;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs,
  Grids, StdCtrls, ExtCtrls;

type
  TfrmListFunc = class(TForm)
    StringGrid1: TStringGrid;
    Panel1: TPanel;
    Label1: TLabel;
    procedure FormCreate(Sender: TObject);
    procedure FormResize(Sender: TObject);
  private
    { Private declarations }
    function ListToStringGrid(AFileName: String): Integer;
  public
    { Public declarations }
  end;

var
  frmListFunc: TfrmListFunc;

implementation

{$R *.DFM}

procedure TfrmListFunc.FormCreate(Sender: TObject);
var Total : Integer;
begin
  Total := ListToStringGrid('List_Functions.txt');
  Label1.Caption := 'Function list: Count = '+IntToStr(Total) + '  (not counting overloaded functions)';
end;

function TfrmListFunc.ListToStringGrid(AFileName: String): Integer;
var f: TextFile;
  tmpStr: String;
  i,j, Len: Integer;
begin
  Result := 0;
  With StringGrid1 do
  begin
    RowCount :=1;
    Cells[0,0] := '(Unit) Name';
    Cells[1,0] := 'Description';
  end;
  i := 0;
  If FileExists(AFileName) then
  begin
    AssignFile(f,AFileName);
    try
      Reset(f);
      While Not Eof(f) do
      begin
        readln(f,tmpStr);
        Inc(i);
        StringGrid1.RowCount := i+1;
        Len := Length(tmpStr);
        j:= Pos(Chr(9),tmpStr);
        StringGrid1.Cells[0,i] := Copy(tmpStr,0,j-1);
        StringGrid1.Cells[1,i] := Copy(tmpStr,j+1,len-j);
        if StringGrid1.Cells[1,i]<>'' then Result := Result +1;
      end;
      StringGrid1.FixedRows := 1;
    finally
      CloseFile(f);
    end;
  end;
end;

procedure TfrmListFunc.FormResize(Sender: TObject);
begin
  StringGrid1.ColWidths[1] := StringGrid1.Width - StringGrid1.ColWidths[0];
end;

initialization
  RegisterClass(TfrmListFunc);

end.

File: Stats.CPP/List_Func.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "List_Func.h"
#include "FileCtrl.hpp"
#include "Basic_Form.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TfrmListFunc *frmListFunc;
//---------------------------------------------------------------------------
__fastcall TfrmListFunc::TfrmListFunc(TComponent* Owner)
	: TForm(Owner)
{
	int Total = ListToStringGrid("List_Functions.txt");
	Label1->Caption = "Function list: Count = "+IntToStr(Total) + "  (not counting overloaded functions)";
}
//---------------------------------------------------------------------------
int __fastcall TfrmListFunc::ListToStringGrid(UnicodeString filename)
{
	int i;
	int res = 0;

	String dir = CodePath();
	if (!FileExists(dir + filename))
	{
		if (CodePathSelect())
		{
			dir = CodePath();
		}
	}
	if (!FileExists(dir + filename)) return 0;
	TStrings *s = new TStringList();
	try {
		s->LoadFromFile(dir + filename);
		StringGrid1->RowCount =1;
		StringGrid1->Cells[0][0] = "(Unit) Name";
		StringGrid1->Cells[1][0] = "Description";
		for (i = 0; i<s->Count; i++) {
			StringGrid1->RowCount = i+1;
			int j = Pos((char)9,(*s)[i]);
			StringGrid1->Cells[0][i+1] = (*s)[i].SubString(0,j-1);
			StringGrid1->Cells[1][i+1] = (*s)[i].SubString(j+1,(*s)[i].Length() - j);
			if (StringGrid1->Cells[1][i+1]!="") res++;
		}
	}
	__finally
	{
		delete s;
	}
	StringGrid1->FixedRows = 1;
	return res;
}
void __fastcall TfrmListFunc::FormResize(TObject *Sender)
{
	StringGrid1->ColWidths[1] = StringGrid1->Width - StringGrid1->ColWidths[0];
}
//---------------------------------------------------------------------------

